src/parse/edit_data.jl

```diff
@@ -604,36 +605,40 @@ end
         - `overwrite::Bool = false`: If data in the target region `r.second` is already present,
             should it be overwritten?
 
+
+
 # Returns:
 - `df::DataFrame` with only the desired keys.
 """
 function filter_with(
     df::DataFrame,
     set::Any;
+    drop::Bool = false,
     extrapolate::Bool = false,
     forward::Bool = true,
     backward::Bool = true,
     r::Pair = "md" => "dc",
     overwrite::Bool = false
 )
-    df_ans = copy(df)
-    cols_ans = propertynames(df_ans)
+    cols = propertynames(df)
 
     # Find keys that reference both column names in the input DataFrame df and
     # values in the set Dictionary. Then, created a DataFrame containing all permutations.
-    cols = find_oftype(df, Not(AbstractFloat))
-    cols_set = intersect(cols, collect(keys(set)))
+    cols_key = find_oftype(df, Not(AbstractFloat))
+    cols_set = intersect(cols_key, collect(keys(set)))
     vals_set = [intersect(unique(df[:,k]), ensurearray(set[k])) for k in cols_set]
-    # vals_set = [set[k] for k in cols_set]
 
     # Drop values that are not in the current set.
     df_set = DataFrame(permute(NamedTuple{Tuple(cols_set,)}(vals_set,)))
-    df_ans = innerjoin(df_ans, df_set, on = cols_set)
+    df = innerjoin(df, df_set, on = cols_set)
     
     if extrapolate
-        :yr in cols_set && (df_ans = extrapolate_year(df_ans, set; forward = forward, backward = backward))
-        :r in cols_set  && (df_ans = extrapolate_region(df_ans, r; overwrite = overwrite))
+        :yr in cols_set && (df = extrapolate_year(df, set; forward = forward, backward = backward))
+        :r in cols_set  && (df = extrapolate_region(df, r; overwrite = overwrite))
     end
-    
-    return sort(df_ans[:,cols_ans])
+
+    # If one of the filtered DataFrame columns contains only one unique value, drop it.
+    drop && setdiff!(cols, cols_set[length.(unique.(eachcol(df[:,cols_set]))) .=== 1])
```

Comment:
@jonmbecker here's the line that adds the dropping feature to `filter_with`

Reply:
Great, thank you. I think this is also a solution if :yr is prefiltered
```
df = read_data_temp("ys0",mod_year,data_temp_dir,"Sectoral supply")
d = convert_type(Dict, df; drop_cols=[:yr], value_col = :Val)
```